src/platforms/esp32/components/avm_builtins/gpio_driver.c

```diff
@@ -89,27 +133,15 @@ struct GPIOData
     struct ListHead gpio_listeners;
 };
 
-static inline term gpio_set_pin_mode(term gpio_num_term, term mode)
+static inline term gpio_set_pin_mode(Context *ctx, term gpio_num_term, term mode_term)
 {
     int gpio_num = term_to_int(gpio_num_term);
 
-    esp_err_t result;
-    switch (mode) {
-        case INPUT_ATOM:
-            result = gpio_set_direction(gpio_num, GPIO_MODE_INPUT);
-            break;
-
-        case OUTPUT_ATOM:
-            result = gpio_set_direction(gpio_num, GPIO_MODE_OUTPUT);
-            break;
-
-        case OUTPUT_OD_ATOM:
-            result = gpio_set_direction(gpio_num, GPIO_MODE_OUTPUT_OD);
-            break;
-
-        default:
-            return ERROR_ATOM;
+    gpio_mode_t mode = interop_atom_term_select_int(ctx->global, pin_mode_table, mode_term);
+    if (UNLIKELY(mode < 0)) {
+        return ERROR_ATOM;
     }
+    esp_err_t result = gpio_set_direction(gpio_num, GPIO_MODE_OUTPUT);
```

Comment:
Shouldn't this be `gpio_set_direction(gpio_num, mode);`?  (What happened to `GPIO_MODE_INPUT ` and `GPIO_MODE_OUTPUT_OD `?)

Reply:
Take a look to `pin_mode_table`, they should be all there.